File: src/1436. Destination City/1436.py
from typing import List
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        citiesA = {path[0] for path in paths}
        return next(path[1] for path in paths if path[1] not in citiesA)
    def destCity(self, paths: List[List[str]]) -> str:
        return (set((city:=list(zip(*paths)))[1]) - set(city[0])).pop()

File: src/1436. Destination City/1436.go
func destCity(paths [][]string) string {
    citiesA := map[string]bool{}
    for _, path := range paths {
        citiesA[path[0]] = true
    }
    for _, path := range paths {
        if !citiesA[path[1]] {
            return path[1]
        }
    }
    return ""
}

